File: 1656-count-good-triplets/count-good-triplets.py
class Solution:
    def countGoodTriplets(self, arr: List[int], a: int, b: int, c: int) -> int:
        cnt = 0
        n = len(arr)
        i, j, k = 0, 1, 2

        while i < n - 2:
            if j >= n - 1:
                i += 1
                j = i + 1
                k = j + 1
                continue
            if k >= n:
                j += 1
                k = j + 1
                continue

            if abs(arr[i] - arr[j]) <= a and abs(arr[j] - arr[k]) <= b and abs(arr[i] - arr[k]) <= c:
                cnt += 1

            k += 1

        return cnt

File: 1656-count-good-triplets/count-good-triplets.java
class Solution {
    public int countGoodTriplets(int[] arr, int a, int b, int c) {
        int cnt = 0;
        int n = arr.length;
        int i = 0, j = 1, k = 2;

        while (i < n - 2) {
            if (j >= n - 1) {
                i++;
                j = i + 1;
                k = j + 1;
                continue;
            }

            if (k >= n) {
                j++;
                k = j + 1;
                continue;
            }

            if (Math.abs(arr[i] - arr[j]) <= a &&
                Math.abs(arr[j] - arr[k]) <= b &&
                Math.abs(arr[i] - arr[k]) <= c) {
                cnt++;
            }

            k++;
        }

        return cnt;
    }
}
